src/Microsoft.TemplateEngine.Edge/TemplateUpdates/InstallUnitDescriptorFactory.cs

```diff
@@ -0,0 +1,44 @@
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.Abstractions.Mount;
+using Microsoft.TemplateEngine.Abstractions.TemplateUpdates;
+using Microsoft.TemplateEngine.Abstractions;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Edge.TemplateUpdates
+{
+    // These are written with the assumption that a mount point can only deal with one type of template pack.
+    public static class InstallUnitDescriptorFactory
+    {
+        // For reading an existing descriptor.
+        public static bool TryParse(IEngineEnvironmentSettings environmentSettings, JObject jobject, out IInstallUnitDescriptor descriptor)
+        {
+            foreach (IInstallUnitDescriptorFactory factory in environmentSettings.SettingsLoader.Components.OfType<IInstallUnitDescriptorFactory>().ToList())
```

Comment:
Since we're already asserting that the configuration is a JObject, we could modify it on write to have the Id of the factory that can read it. With that, we'd be able to cache the set of factories and look up the one we need by Id rather than having to loop over them to see which one says it handles it.

Reply:
I thought about doing that, but it would be opinionated in that all descriptor serializations have the factoryId as a top-level key. If we want to enforce that, I'm all for what you suggested.